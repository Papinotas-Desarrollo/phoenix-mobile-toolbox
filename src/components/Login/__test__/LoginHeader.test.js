import * as React from 'react';
import { shallow } from 'enzyme';
import { Image } from 'react-native';
import { LoginHeader } from '../';

describe('Snapshot test', () => {
  it('No props snapshot', () => {
    const tree = shallow(<LoginHeader />);
    expect(tree).toMatchSnapshot();
    expect(tree.children().find(Image).length).toBe(0);
  });
  it('With props snapshot', () => {
    const source = require('../../../assets/papinotascolegio.png');
    const props = {
      source,
    };
    const tree = shallow(<LoginHeader {...props} />);
    expect(tree).toMatchSnapshot();
    expect(
      tree
        .children()
        .find(Image)
        .props().source.testUri
    ).toBe(source.testUri);
  });
});
