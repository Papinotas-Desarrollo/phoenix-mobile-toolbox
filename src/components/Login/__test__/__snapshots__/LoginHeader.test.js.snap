// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test No props snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <View />,
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#4A90E2",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "minHeight": 70,
        "paddingHorizontal": 10,
        "paddingVertical": 20,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <View />,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#4A90E2",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "minHeight": 70,
          "paddingHorizontal": 10,
          "paddingVertical": 20,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Snapshot test With props snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginHeader
    source={
      Object {
        "testUri": "../../../src/assets/papinotascolegio.png",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Component
        source={
          Object {
            "testUri": "../../../src/assets/papinotascolegio.png",
          }
        }
      />,
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#4A90E2",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "minHeight": 70,
        "paddingHorizontal": 10,
        "paddingVertical": 20,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "source": Object {
          "testUri": "../../../src/assets/papinotascolegio.png",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component
          source={
            Object {
              "testUri": "../../../src/assets/papinotascolegio.png",
            }
          }
        />,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#4A90E2",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "minHeight": 70,
          "paddingHorizontal": 10,
          "paddingVertical": 20,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "source": Object {
            "testUri": "../../../src/assets/papinotascolegio.png",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
